### ISSUE TITLE
Username/Password authentication: Password not saved if "Test" Button not pressed

### ISSUE THREAD

**Issue by Structed** [SouthbankSoftware/dbkoda#26]
Affected Version: dbKoda 1.0.1
Operating System: Windows 10 Pro 1803, Build 17134.165

# Steps to Reproduce
1.  Open dbKoda
2. Enable "Password Store" on the Home Screen
3. Click the "+" Button on the "Connection Profiles" sidebar
4. Check the checkbox "SCRAM-SHA-1(username/password)"
5. Fill out the connection profile data, including username and password
6. Click the "Save" Button

## Expected Behaviour
* Profile will be saved, including the Password in the "Password Store"
* When trying to connect, and the connection parameters are valid, dbKoda should connect to the specified server/db

## Actual Behavior
* dbKoda does not connect, reporting that there is no password stored

## Workaround
Press the "Test" Button before saving

**Comment by michaeljharrison** [comment]
Hey Structed,

Thanks again for submitting this issue so clearly for us. At the moment we only save to the password store once a valid connection is confirmed, this is what's causing the issue.
I've put an item on the backlog to make sure the password is stored when the "save" button is clicked as well.

Thanks,
Mike from dbKoda.

**Comment by Structed** [comment]
I understand the motivation and - for me at least, this is also perfectly fine. Perhaps I should have said, that a simple disable of the save button or a UI notification would be enough for the user to u derstand that dbKoda requires a test of the connection before saving it. Why would you save a connection if you're not sure it works? I just kind of accidentally hit the save instead of the test button.

**Comment by michaeljharrison** [comment]
Yeah I totally agree, it's definitely a bit confusing for the user, specially since the user won't know it hasn't been saved until they try to connect!
I think we can change the behaviour so the password is saved, but when you try to connect the password will be used, and a new prompt will appear if the password fails but I'll work it through with our UX guy as well and see what he thinks the best solution is.
